### PR TITLE
Morse Code Decoder

### PR DESCRIPTION
**Pull request by Baayeh**
### In this milestone, we:

- Created a method to decode a Morse code character
- Created a method to decode an entire word in Morse code
- Created a method to decode the entire message in Morse code

### CODE REVIEW

File: decoder.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  words = message.split('   ')
+  decoded_msg = ''
+  words.each do |word|
+    letters = word.split
+    letters.each do |letter|
+      decoded_msg += decode_char(letter)
+    end
+    decoded_msg += ' '
+  end
+  decoded_msg.strip
```
Comment (julie-ify) [inline decoder.rb]: - [x] You are required to use the `decode_word` method here not `decode_char`. Kindly refactor this.

  Hint:
  ```
  def decode(message)
    words = message.split('   ')
    decoded_msg = ''
    words.each do |word|
      decoded_msg += decode_word(word) + " "
    end
    decoded_msg.strip
  end
  ```

File: decoder.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    return key if character == value
+  end
+end
+
+def decode_word(word)
+  split_word = word.split
+  split_word.each do |i|
+    decode_char(i)
+  end
+end
```
Comment (julie-ify) [inline decoder.rb]: - [x] This method returns an array of characters. Kindly fix that.

  didd

  Hint:
  ```
  def decode_word(word)
    split_word = word.split
    word = []
    split_word.each do |i|
      word << decode_char(i)
    end
    word.join
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Baayeh and @j-tee,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decoder.rb (hunk 1 above)
Comment (Baayeh) [reply decoder.rb]: Thank you for the suggestion. It has been refactored

```
def decode(message)
  words = message.split('   ')
  decoded_msg = ''
  words.each do |word|
    decoded_msg += "#{decode_word(word)} "
  end
  decoded_msg.strip
end
```

File: decoder.rb (hunk 2 above)
Comment (Baayeh) [reply decoder.rb]: Thanks for the hint. The code has been refactored

**Review by danmainah — APPROVED** [review]
Hi @Baayeh and @j-tee , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**